### PR TITLE
FIXED #174

### LINKED ISSUE #174: Missing Updates of Bookmarked List
On UnBookmarking a session ,it doesn't get removed from the Bookmarked List , until and unless we change the fragment.
gif_2

### PR DESCRIPTION
**Pull request by opticod**
#174

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/fragments/BookmarksFragment.java [hunk 1]
```diff
@@ -29,11 +29,17 @@
  * Created by manan on 22-05-2015.
  */
 public class BookmarksFragment extends Fragment {
-    SessionsListAdapter sessionsListAdapter;
+    static SessionsListAdapter sessionsListAdapter;
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/fragments/BookmarksFragment.java]: why is the adapter variable static?

Comment (opticod) [inline app/src/main/java/org/fossasia/openevent/fragments/BookmarksFragment.java]: I am referencing `notifyBookmarkedChanged()` from static context of  `onOptionsItemSelected(MenuItem item)` ..so need to keep `sessionsListAdapter` static.

File: app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java [hunk 2]
```diff
@@ -124,6 +124,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
                     createNotification();
                 }
                 sendBroadcast(new Intent(BookmarkWidgetProvider.ACTION_UPDATE));
+                BookmarksFragment.notifyBookmarkedChanged();
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/activities/SessionDetailActivity.java]: why is bookmarks fragment being accessed from here?

File: app/src/main/java/org/fossasia/openevent/fragments/BookmarksFragment.java (hunk 1 above)
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/fragments/BookmarksFragment.java]: nope this isn't right, firstly, `SessionDetailActivity` should not access the Bookmarks fragment plus, you shouldn't need to access cached objects like this.

**Comment by creativepsyco** [comment]
I think the right way to fix this is to be just refresh the `BookmarksFragment` when it is shown. You cannot depend on Bookmarks fragment being available because it is in the activity behind the current one (SessionsDetail), also trying to fix it via `static` modifiers makes it not garbage collected when it should be (Static members are assigned on the class address space, not on the object address space) See http://stackoverflow.com/questions/453023/are-static-fields-open-for-garbage-collection

**Comment by opticod** [comment]
@creativepsyco done.
Thanks for pointing out..this all helps a lot and to improve while contributing.. :smiley:

File: app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java [hunk 3]
```diff
@@ -94,6 +98,24 @@ public void moveItem(int fromPosition, int toPosition) {
         notifyItemMoved(fromPosition, toPosition);
     }
 
+    public void refresh() {
```
Comment (creativepsyco) [inline app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: SessionsListAdapter has no knowledge of the DbSingleton now here you are adding some more coupling.

Comment (opticod) [inline app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: Thanks, then simply we can have it in Bookmarks fragment only....and after refreshing the list, we will call `adapter.notifyDataSetChanged()`

**Pull request merged**